### PR TITLE
[DSv4] Enable paged indexer for prefill

### PR DESCRIPTION
**Pull request by gau-nernst**
## Purpose

This PR enables paged indexer path prefill i.e. use DeepGEMM's `fp8_fp4_paged_mqa_logits` instead of `fp8_fp4_mqa_logits` for prefill. This is only available for SM100. It skips gather K kernel (`cp_gather_indexer_k_quant_cache`) -> faster.

However, it materializes the full `Q @ K.T` logits, which has the size of `max_prefill_tokens x max_compressed_model_len`. Using 8192 tokens, 1M context, and compress ratio 4, the logits take `8192 * (1048576 / 4) * 4 bytes = 8 GiB`. Due to this reason, initially I also implemented a chunk version of paged indexer (still available in `benchmarks/kernels/benchmark_prefill_sparse_indexer.py`). But in e2e benchmark, full paged indexer (no chunking) actually has **lower** peak memory usage. I'm not sure why this is the case... Hence, I only keep full paged prefill in this PR (chunking logic is still in the git history and the benchmark script, so we can still revisit it if needed).

This feature will be gated behind the flag `--enable-sparse-indexer-paged-prefill` so that it doesn't change the default behavior.

## Microbenchmarks

This includes time for topk kernel as well. For the current "flat" kernels (`fp8_fp4_mqa_logits`), the timing also includes K gather op.

### FP8

```
python benchmarks/kernels/benchmark_prefill_sparse_indexer.py --dtype fp8 --max_query_len 4096 --max_context_len 1000 3000 10000 30000 100000 300000 1000000 --log_axis --graph_path fp8_large.png
```

fp8_large

### FP4

```
python benchmarks/kernels/benchmark_prefill_sparse_indexer.py --dtype fp4 --max_query_len 4096 --max_context_len 1000 3000 10000 30000 100000 300000 1000000 --log_axis --graph_path fp4_large.png
```

fp4_large

## Accuracy evals

Eval | W/o paged prefill indexer | W/ paged prefill indexer
---|---|---
GSM8K (strict-match) | 0.9462 | 0.9560
AIME25 | 0.9437 | 0.9583

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by gau-nernst** [comment]
There was a bug in benchmark script. Paged version is actually much slower than current implementation. Closing for now.
